### DISCUSSION TITLE
RHF + CleaveJS  reset field after submit form

### DISCUSSION THREAD

**Discussion by thiagofernando** [react-hook-form/react-hook-form#2727] [Q&A]
Hi, I'm using RHF with Cleave and MUI. I am unable to clear the field after submitting the form. Does anyone have any tips?

https://codesandbox.io/s/rhf-mui-cleave-d8r26

**Comment by bluebill1049** [comment]
can you take a look this example here: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r ? probably you are missing wire up the value and methods.

**Comment by thiagofernando** [comment]
i already tried passing value , onChange and without them. I can't get it to work

**Reply by bluebill1049** [reply]
no, you should use render, pass me your example in a CSB.

**Reply by thiagofernando** [reply]
this one https://codesandbox.io/s/rhf-mui-cleave-d8r26

**Reply by bluebill1049** [reply]
<img width="497" alt="Screen Shot 2020-08-28 at 9 03 02 am" src="https://user-images.githubusercontent.com/10513364/91503217-49c95780-e90d-11ea-8406-529339212273.png">
why are you using setValue in controller? u should use onChange method from render prop. take a look at the doc for more refernece.

**Reply by thiagofernando** [reply]
I'm using setValue to get RawValue instead of the formatted value. However, even without it, the form reset does not work.

**Reply by bluebill1049** [reply]
fix your codesandbox first, i will take a look again.

**Reply by thiagofernando** [reply]
I understand that it is not wrong, because I want the form to send rawValue. In any case I commented this line, onChange is now being sent with the other props

**Reply by bluebill1049** [reply]
if you want the raw input, i suggest you create a hidden input and register that and setValue, you shouldn't manipulate the data at CleaveJs level.

**Reply by thiagofernando** [reply]
Understood. However, I still can't reset the field with the Cleave after sending the form. Can you check the CSB?
